src/engine/sys/sdl_glimp.cpp

```diff
@@ -2285,77 +2290,98 @@ static void GLimp_InitExtensions()
 	// made required in OpenGL 4.3
 	glConfig2.computeShaderAvailable = LOAD_EXTENSION_WITH_TEST( ExtFlag_NONE, ARB_compute_shader, r_arb_compute_shader.Get() );
 
-	// Some of the mesa 24.x driver versions have a bug in their shader compiler related to bindless textures,
-	// which results in either glitches or the shader compiler crashing (when material system is enabled)
-	if ( glConfig2.driverVendor == glDriverVendor_t::MESA )
 	{
-		bool foundMesa241 = false;
+		bool bindlessTextureEnabled = r_arb_bindless_texture.Get();
 
-		if ( workaround_glDriver_mesa_v241_disableBindlessTexture.Get() )
+		if ( bindlessTextureEnabled && glConfig2.hardwareVendor == glHardwareVendor_t::ATI )
 		{
-			std::string mesaVersion = "";
+			/* Some of the mesa 24.x driver versions have a bug in their shader compiler
+			related to bindless textures, which results in either glitches or the shader
+			compiler crashing (when material system is enabled). */
+			if ( glConfig2.driverVendor == glDriverVendor_t::MESA )
+			{
+				bool foundMesa241 = false;
+				std::string mesaVersion = "";
 
-			static const std::vector<std::pair<std::string, std::vector<std::string>>> versions = {
-				{ "1.0", { "1.0-devel", "1.0-rc1", "1.0-rc2", "1.0-rc3", "1.0-rc4", "1.0", } },
-				{ "1.",  { "1.1", "1.2", "1.3", "1.4", } },
-				{ "2.0", { "2.0-devel", "2.0-rc1", } },
-			};
+				static const std::pair<std::string, std::vector<std::string>> versions[] = {
+					{ "1.0", { "1.0-devel", "1.0-rc1", "1.0-rc2", "1.0-rc3", "1.0-rc4", "1.0", } },
+					{ "1.",  { "1.1", "1.2", "1.3", "1.4", } },
+					{ "2.0", { "2.0-devel", "2.0-rc1", } },
+				};
 
-			for ( auto& vp : versions ) {
-				mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
+				for ( auto& vp : versions )
+				{
+					mesaVersion = Str::Format( "Mesa 24.%s", vp.first );
```

Comment:
Is there any reason not to just put all the more complete version numbers (`1.0-devel`, `1.0-rc1` etc.) into one array and check against them? The nested loop here seems redundant.

Reply:
We can iterate all the numbers indeed, but I thought it would be faster to not test all strings so I went the divide & conquer way. I can make it simpler if considered better.

Reply:
I rewrote the code to walk the string and not retest what is already tested. I kept the dichotomous search, and I updated the known version strings.

Reply:
So for matching `24.2.0` it matches `Mesa 24.`, then `2.` then `0`.